pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/store/SegmentLocalFSDirectory.java

```diff
@@ -349,6 +356,42 @@ public boolean hasIndexFor(String column, ColumnIndexType type) {
       return _columnIndexDirectory.hasIndexFor(column, type);
     }
 
+    public boolean hasSegmentIndex(SegmentIndexType type) {
+      if (type != SegmentIndexType.STAR_TREE_INDEX) {
+        throw new IllegalArgumentException("Unknown SegmentIndexType: " + type);
+      }
+      return _starTreeIndexReader != null;
+    }
+
+    public SegmentDirectory.Reader getSegmentIndexReaderFor(String indexName, SegmentIndexType type) {
```

Comment:
I don't think this is a common abstraction for segment level index. Suggest for now just add APIs for star-tree: `getStarTreeReader(int starTreeId)`